builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1556,11 +1494,5 @@ func (vm *virtualMachine) deployVirtualMachine(c *govmomi.Client) error {
 
 	newVM.PowerOn(context.TODO())
 
-	ip, err := newVM.WaitForIP(context.TODO())
```

Comment:
Same here. WaitForNetIP will be done in resourceVSphereVirtualMachineRead, which is called at the end.